tests/test_backend/test_mlir/test_mlir_compiler.py

```diff
@@ -0,0 +1,7 @@
+import magma as m
+from examples import complex_when_lazy_tuple
+
+
+def test_mlir_compiler_finalize_when_order():
+    m.compile("build/complex_when_lazy_tuple", complex_when_lazy_tuple,
+              output="mlir")
```

Comment:
nit
```suggestion
    m.compile(
        "build/complex_when_lazy_tuple",
        complex_when_lazy_tuple,
        output="mlir",
    )
```

also let's use a tempdir and kill the build dir and .gitignore